reana-ui/src/pages/workflowDetails/WorkflowDetails.js

```diff
@@ -45,20 +45,20 @@ function WorkflowDetails() {
     dispatch(fetchWorkflow(workflowId));
   }, [dispatch, workflowId]);
 
-  if (loading || !workflow) {
+  if (loading) {
     return (
       <Dimmer active>
         <Loader>Loading workflow...</Loader>
       </Dimmer>
     );
   }
 
-  if (_.isEmpty(workflow)) {
+  if (!workflow) {
     return (
       <Container text className={styles.warning}>
         <Message
           icon="warning sign"
-          header="Workflow does not exist."
+          header="Sorry you are not authorised to see this workflow or does not exist."
```

Comment:
"or it does not..."

Perhaps better said in the inverse order, "Sorry, this workflow either does not exist or you are not authorised to see it." I'll amend the PR accordingly.

P.S. In the future we should split these two cases, as it might cause confusion...